### PR TITLE
Upgrade to Go 1.15

### PR DESCRIPTION
**Pull request by kakkoyun**
Signed-off-by: Kemal Akkoyun <kakkoyun@gmail.com>

* [ ] I added CHANGELOG entry for this change.

https://golang.org/doc/go1.15#compiler
> Go 1.15 reduces typical binary sizes by around 5% compared to Go 1.14 by eliminating certain types of GC metadata and more aggressively eliminating unused type metadata.

#### 1.14.7
Screenshot 2020-08-12 at 10 36 25

#### 1.15.0
Screenshot 2020-08-12 at 10 36 42

## Changes

* Upgrade use go version
* Upgrade `promu` version
* Upgrade CI go version

## Verification

* `make build`
* `make test-local`

### CODE REVIEW

**Comment by kakkoyun** [comment]
Somehow Netlify builds use Go and jobs fail. I couldn't figure where it's specified.

**Comment by kakkoyun** [comment]
Found the reason of Netlify failures https://github.com/netlify/build-image/blob/450c24b893036c2813a0b0d0a2a05bcdd81935a4/run-build-functions.sh
Netlify uses https://github.com/netlify/gimme to install Go and `gimme` Go version is listed as `1.15` which is specified in `.go-version`

File: .dockerignore [hunk 1]
```diff
@@ -0,0 +1,31 @@
+/prometheus
+/thanos
+/vendor/
+/mixin/vendor/
+node_modules/
+
+# Ignore minikube setup working dirs.
+kube/bin
+kube/.kube
+kube/.minikube
```
Comment (yeya24) [inline .dockerignore:10]: Do we use Kubernetes in our tests?

Comment (kakkoyun) [reply .dockerignore:10]: I have copied those from `.gitignore`. I haven't checked them to be honest.

**Comment by kakkoyun** [comment]
There's a failing test. I need to check the underlying cause.
```
--- FAIL: TestRoundTripSplitIntervalMiddleware (0.00s)
    --- FAIL: TestRoundTripSplitIntervalMiddleware/split_to_2_requests (0.00s)
        roundtrip_test.go:269: roundtrip_test.go:269:
            
            	exp: 2
            
            	got: 1
```

**Comment by yeya24** [comment]
> There's a failing test. I need to check the underlying cause.
> 
> ```
> --- FAIL: TestRoundTripSplitIntervalMiddleware (0.00s)
>     --- FAIL: TestRoundTripSplitIntervalMiddleware/split_to_2_requests (0.00s)
>         roundtrip_test.go:269: roundtrip_test.go:269:
>             
>             	exp: 2
>             
>             	got: 1
> ```

@kakkoyun Reported here https://github.com/thanos-io/thanos/issues/3084.

I am checking that now. Please ignore this first.

**Review by bwplotka — APPROVED** [review]
Thank you! LGTM!

**Pull request merged**